### PR TITLE
Incorrect phasing in the noise-gen script

### PR DESCRIPTION
**Pull request by colemanalan**
Issue being addressed:
Solves issue #699

### CODE REVIEW

File: NuRadioReco/utilities/noise.py [hunk 1]
```diff
@@ -102,10 +102,7 @@ def rolled_sum_slicing(traces, rolling):
     sumtr = traces[0].copy()
     for i in range(1, len(traces)):
         r = rolling[i]
-        if r > 0:
-            sumtr[:-r] += traces[i][r:]
-            sumtr[-r:] += traces[i][:r]
-        elif r < 0:
+        if r != 0:
```
Comment (shallmann) [inline NuRadioReco/utilities/noise.py:105]: Thanks for spotting and correcting. That does the correct thing now... 

A simple test would have avoided this in the first place :/
I verified it works for rolling<trace length...that should not happen, but one could put
`r = rolling[i]%len(sumtr)` for completeness (with that, `r` would never have been negative in the first place)

I used this to test:
```
def test_rolling_functions(n_tests=100, n_traces=5):
    """ tests for equality of output of `rolled_sum_roll` and `rolled_sum_slicing` """
    traces = [np.random.randint(-10,11,100) for i in range(n_traces)]

    for i in range(n_tests):
        rollings = np.random.randint(-100,100,n_traces)
        rollings -= rollings[0]

        assert np.alltrue(rolled_sum_roll(traces, rollings) == rolled_sum_slicing(traces, rollings)), "not equal"
```

File: NuRadioReco/utilities/noise.py [hunk 2]
```diff
@@ -331,7 +328,8 @@ def __init__(self, detector_filename, station_id, triggered_channels,
             delays -= np.max(delays)
 
             roll = np.array(np.round(np.array(delays) * self.sampling_rate * self.upsampling)).astype(int)
-            self.beam_time_delays[iBeam] = roll
+            # subtract off the delay of antenna 0 because `rolled_sum_slicing` assumes this is the case
+            self.beam_time_delays[iBeam] = roll - roll[0]
```
Comment (shallmann) [inline NuRadioReco/utilities/noise.py:332]: yes perfect

**Review by shallmann — APPROVED** [review]

File: NuRadioReco/utilities/noise.py (hunk 1 above)
Comment (colemanalan) [reply NuRadioReco/utilities/noise.py:105]: Thanks for the double check!

File: changelog.txt [hunk 3]
```diff
@@ -29,6 +29,7 @@ can then iterate over a channel group, which yields all channels for a given gro
 loaded again when reading in the Detector from a nur file
 - Added LOFAR coordinates to Detector site coordinates
 - Implemented mattak dataset iterator in readRNOGDataMattak.run()
+- Fixed bug in phased-array noise generation script
```
Comment (cg-laser) [inline changelog.txt:32]: can you specify what exactly was wrong? Something like: "Previous version always generated triggered noise that triggered in half of the beams"

**Review by shallmann — APPROVED** [review]
Fine for me.

**Pull request merged**